packages/data-point/lib/reducer-types/reducer-list/resolve.js

```diff
@@ -9,24 +7,22 @@ const utils = require("../../utils");
  * @param {ReducerList} reducerList
  * @returns {Promise}
  */
-function resolve(manager, resolveReducer, accumulator, reducerList) {
+async function resolve(manager, resolveReducer, accumulator, reducerList) {
   const reducers = reducerList.reducers;
   if (reducers.length === 0) {
-    return Promise.resolve(undefined);
+    return undefined;
   }
 
-  const initialValue =
-    accumulator.value === undefined ? null : accumulator.value;
-  const result = Promise.reduce(
-    reducers,
-    (value, reducer) => {
-      const itemContext = utils.set(accumulator, "value", value);
-      return resolveReducer(manager, itemContext, reducer);
-    },
-    initialValue
-  );
+  let value = accumulator.value === undefined ? null : accumulator.value;
+
+  for (let index = 0; index < reducers.length; index += 1) {
```

Comment:
Is there any reason why you're using a for loop instead of a more-fp forEach?
I've seen it in other files too, but in those cases, I think you wanted to break the loop.

Reply:
yes, this code needs to be executed serially, as in one after the other, the output of one reducer, is the input of the next, so we need to execute each reducer, await for it, and then move on the next reducer. 

to make this more fp we could write an asynchronous `composition` method, but internally at the end would be this again. 

another option is to use a reducer/then pattern: 

```js
functions.reduceRight(
  (chain, func) => chain.then(func),
  Promise.resolve(input)
)
```

But I find that harder to read than a simple for loop